### ISSUE TITLE
Consider removing the `Eager` versions from `baresult`

### ISSUE THREAD

**Issue by kdungs** [kdungs/go-result#4]

**Comment by kdungs** [comment]
The whole reason to introduce these helpers and treat `(T, error)` as a check monad is to improve ergonomics. With the eager versions, calling code has to always store intermediates because we're only allowed to wrap the result of a function call when it's the only argument. For example.

```go
processed, err := baresult.Fmap(ProcessData)(os.ReadFile(filePath))
```

vs.

```go
data, err := os.ReadFile(filePath)
processed, err := baresult.EagerFmap(ProcessData, data, err)
```

For the same reason, the eager versions also don't allow chaining. Come to think of it, we probably don't need the eager versions, at all. Best case scenario: #1 yields that the functions are inlined anyway 🤷 .

**Comment by kdungs** [comment]
Fixed by f6b1bad3b6c333c447fbbba167867b201599e23a
